YoYoProject/Controllers/GMResource.cs

```diff
@@ -11,5 +12,11 @@ public abstract class GMResource : ControllerBase
         internal Guid ResourceInfoId { get; set; }
 
         internal abstract void Create(string name);
+
+        public void AddResourceToFolder(string filtername)
+        {
+            var fld = Project.Resources.GetAllOfType<GMFolder>().Where(n => n.FilterType == filtername);
+            fld.FirstOrDefault().Children.Add(this);
```

Comment:
Multiple possible null reference exceptions here.
FirstOrDefault() should just be First() if you're not going to handle the null.